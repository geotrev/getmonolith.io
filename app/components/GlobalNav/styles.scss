@import "app/global";

@mixin align-nav-item($justify) {
  justify-content: center;

  @include breakpoint-up(small) {
    justify-content: $justify;
  }
}

#global-nav {
  background: $brand-dark-blue;
  padding: 0;
  z-index: 5;

  .nav-list {
    list-style-type: none;
    margin: 0;
    padding: 0;

    .logo {
      opacity: 1;

      img {
        height: 30px;
        width: auto;
      }
    }

    > li:first-child {
      @include align-nav-item(flex-start);
      display: inline-flex;
      align-items: center;
    }

    > li:last-child {
      ul {
        @include align-nav-item(flex-end);
        display: flex;
        height: 100%;
        list-style-type: none;
        margin: 0;
        padding: 0;
      }

      li {
        display: flex;
        align-items: center;
      }
    }
  }

  li {
    padding: 0;
  }

  li a {
    @include colored-link($white);
    display: flex;
    text-decoration: none;
    padding: 1rem;
  }
}
