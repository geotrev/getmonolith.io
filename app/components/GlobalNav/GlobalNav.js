import React from "react"
import Github from "react-feather/dist/icons/github"
import Twitter from "react-feather/dist/icons/twitter"
import { Link } from "react-router-dom"
import PropTypes from "prop-types"

import { introductionPath } from "app/routes"
import Logo from "app/assets/images/un-logo.png"
import "./styles.scss"

export default function GlobalNav(props) {
  const handleClick = event => {
    event.preventDefault()
    props.handleRefocusClick(props.mainRef)
  }

  return (
    <nav id="global-nav" className="fluid grid">
      <ul className="nav-list row has-no-padding">
        <li className="is-small-5 is-xsmall-12 column" role="none">
          <Link to="/" className="logo" role="listitem">
            <img src={Logo} alt="Undernet" />
          </Link>
        </li>
        <li className="is-small-7 is-xsmall-12 column">
          <ul className="row">
            <li role="none">
              <button
                role="listitem"
                className="is-visually-hidden-focusable"
                onClick={handleClick}
                type="button"
              >
                Skip to main content
              </button>
            </li>
            <li role="none">
              <a className="has-feather" href="https://www.twitter.com/gwtrev" role="listitem">
                <Twitter role="presentation" focusable="false" />
                <span className="is-visually-hidden">Open link to www.twitter.com/gwtrev</span>
              </a>
            </li>
            <li role="none">
              <a
                className="has-feather"
                href="https://www.github.com/geotrev/undernet"
                role="listitem"
              >
                <Github role="presentation" focusable="false" />
                <span className="is-visually-hidden">
                  Open link to www.github.com/geotrev/undernet
                </span>
              </a>
            </li>
            <li role="none">
              <a href={introductionPath} role="listitem">
                Documentation
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </nav>
  )
}

GlobalNav.propTypes = {
  handleRefocusClick: PropTypes.func.isRequired,
  mainRef: PropTypes.shape({
    current: PropTypes.instanceOf(Element),
  }),
}
