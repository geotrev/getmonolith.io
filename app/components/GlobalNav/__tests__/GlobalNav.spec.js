import React from "react"
import GlobalNav from "../GlobalNav"
import { BrowserRouter as Router } from "react-router-dom"

describe("<Nav />", () => {
  const baseProps = {
    handleRefocusClick: jest.fn(),
    mainRef: {
      current: document.createElement("main"),
    },
  }

  const mountComponent = () => {
    return mount(
      <Router>
        <GlobalNav {...baseProps} />
      </Router>
    )
  }

  it("renders", () => {
    // Given
    const wrapper = mountComponent()
    // Then
    expect(wrapper).toMatchSnapshot()
  })

  it("calls handleRefocusClick prop when hidden button is clicked", () => {
    // Given
    const wrapper = mountComponent()
    // When
    wrapper.find(".is-visually-hidden-focusable").simulate("click")
    // Then
    expect(baseProps.handleRefocusClick).toHaveBeenCalledWith(baseProps.mainRef)
  })
})
