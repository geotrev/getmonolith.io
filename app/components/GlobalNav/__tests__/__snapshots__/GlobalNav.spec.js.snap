// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <GlobalNav
      handleRefocusClick={[MockFunction]}
      mainRef={
        Object {
          "current": <main />,
        }
      }
    >
      <nav
        className="is-fluid grid"
        id="global-nav"
      >
        <ul
          className="nav-list row has-no-p"
        >
          <li
            className="is-sm-5 is-xs-12 column"
            role="none"
          >
            <Link
              className="logo"
              role="menuitem"
              to="/"
            >
              <LinkAnchor
                className="logo"
                href="/"
                navigate={[Function]}
                role="menuitem"
              >
                <a
                  className="logo"
                  href="/"
                  onClick={[Function]}
                  role="menuitem"
                >
                  <img
                    alt="Undernet"
                    src={Object {}}
                  />
                </a>
              </LinkAnchor>
            </Link>
          </li>
          <li
            className="is-sm-7 is-xs-12 column"
          >
            <ul
              className="row"
            >
              <li
                role="none"
              >
                <button
                  className="is-visually-hidden-focusable"
                  onClick={[Function]}
                  role="menuitem"
                  type="button"
                >
                  Skip to main content
                </button>
              </li>
              <li
                role="none"
              >
                <a
                  className="has-feather"
                  href="https://www.twitter.com/gwtrev"
                  role="menuitem"
                >
                  <Twitter
                    focusable="false"
                    role="presentation"
                  />
                  <span
                    className="is-visually-hidden"
                  >
                    Open link to www.twitter.com/gwtrev
                  </span>
                </a>
              </li>
              <li
                role="none"
              >
                <a
                  className="has-feather"
                  href="https://www.github.com/geotrev/undernet"
                  role="menuitem"
                >
                  <Github
                    focusable="false"
                    role="presentation"
                  />
                  <span
                    className="is-visually-hidden"
                  >
                    Open link to www.github.com/geotrev/undernet
                  </span>
                </a>
              </li>
              <li
                role="none"
              >
                <a
                  href="/docs/overview/introduction"
                  role="menuitem"
                >
                  Documentation
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </nav>
    </GlobalNav>
  </Router>
</BrowserRouter>
`;
